app/workers/email_generation_worker.rb

```diff
@@ -1,24 +1,38 @@
 class EmailGenerationWorker
   include Sidekiq::Worker
 
-  def perform(subscription_content_id)
-    subscription_content = SubscriptionContent
-      .includes(:content_change)
-      .includes(subscription: :subscriber)
-      .find(subscription_content_id)
-
-    email = Email.create_from_params!(email_params(subscription_content))
+  def perform
+    while subscription_contents.exists?
+      to_queue = []
 
-    subscription_content.update!(email: email)
+      SubscriptionContent.transaction do
+        subscription_contents.lock.limit(100).find_each do |subscription_content|
+          email = create_email(subscription_content: subscription_content)
+          subscription_content.update!(email: email)
+          to_queue << [email.id, subscription_content.content_change.priority.to_sym]
+        end
+      end
 
-    DeliveryRequestWorker.perform_async_with_priority(
-      email.id, priority: subscription_content.content_change.priority.to_sym,
-    )
+      to_queue.each do |email_id, priority|
+        DeliveryRequestWorker.perform_async_with_priority(email_id, priority: priority)
+      end
```

Comment:
Why do we enqueue at the end of the job instead of during?

Reply:
Because we’re in a transaction here, the emails may not have been committed to the database yet so if we were to trigger the workers before that they wouldn’t be able to find the email.

Reply:
Ahh right. That makes sense.